### PR TITLE
Check language metadata

### PR DESCRIPTION
**Pull request by catreedle**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR introduces `check_language_metadata.py` with the functionality:

- Check and log the differences between the languages in `languages_metadata.json` and `language_data_extraction`.
- Validate that each language and sub-language in `language_metadata.json` has the property `qid` and `iso`.

This code has been tested locally.

### Related issue

- #340

### CODE REVIEW

File: src/scribe_data/check/check_language_metadata.py
Comment (DeleMike) [inline src/scribe_data/check/check_language_metadata.py:14]: @catreedle you can get these from `src/scribe_data/cli/cli_utils.py` - they already exist there

File: src/scribe_data/check/check_language_metadata.py
Comment (DeleMike) [inline src/scribe_data/check/check_language_metadata.py:36]: Also, the `cli_utils.py` has loaded the `language_metadata.json` and `data_type_metadata.json` files for us. You could ignore loading it in here

Comment (DeleMike) [reply src/scribe_data/check/check_language_metadata.py:36]: Screenshot 2024-10-16 at 15 13 42

see the `cli_utils.py` contents

File: src/scribe_data/check/check_language_metadata.py
Comment (DeleMike) [inline src/scribe_data/check/check_language_metadata.py:231]: If this is for a test, then it's fine. but we will be calling this in  `check_project_metadata.yaml` file so no need for this :)

Comment (catreedle) [reply src/scribe_data/check/check_language_metadata.py:231]: thank you! good to know 😊

File: src/scribe_data/check/check_language_metadata.py
Comment (catreedle) [reply src/scribe_data/check/check_language_metadata.py:36]: does it mean we can directly use it? how?

**Comment by andrewtavis** [comment]
One minor thing here, @catreedle: Could we get consistent function docstirng, and ones similar to what we have in this PR in `get_missing_languages`? :)

**Comment by andrewtavis** [comment]
Making the docstring like that means we can use autodoc in the documentation 😊

**Comment by DeleMike** [comment]
hey @catreedle , I'm sorry if I have confused you a bit :(

```python
if __name__ == "__main__":
    check_language_metadata()
```
I just checked and the code you added earlier is quite important if we are gonna be using this file for our workflow. Can you please add it back?

Ty!

**Comment by catreedle** [comment]
> One minor thing here, @catreedle: Could we get consistent function docstirng, and ones similar to what we have in this PR in `get_missing_languages`? :)

on it :)

**Comment by catreedle** [comment]
> hey @catreedle , I'm sorry if I have confused you a bit :(
> 
> ```python
> if __name__ == "__main__":
>     check_language_metadata()
> ```
> 
> I just checked and the code you added earlier is quite important if we are gonna be using this file for our workflow. Can you please add it back?
> 
> Ty!

sure. no worries :)

**Comment by catreedle** [comment]
updated the docstring. let me know if there's amiss @andrewtavis :)

**Review by OmarAI2003 — CHANGES_REQUESTED** [review]

File: src/scribe_data/check/check_language_metadata.py
Comment (OmarAI2003) [reply src/scribe_data/check/check_language_metadata.py:13]: Yes, that approach should work. However, I have a simple suggestion: you might want to download a version of the new JSON file while waiting for the PR to be approved and merged. This way, you can experiment with it and ensure everything works as expected.

**Comment by andrewtavis** [comment]
Quick note being sent to all the testing PRs, if updates are needed now that #402 has been merged, then it'd be great to get those updates to the branch :) If no updates are needed, then let me know 😊

**Comment by catreedle** [comment]
> Quick note being sent to all the testing PRs, if updates are needed now that #402 has been merged, then it'd be great to get those updates to the branch :) If no updates are needed, then let me know 😊

Hi! I've made the necessary updates after the #402 merge. Everything should be good to go now 😊.

File: src/scribe_data/check/check_language_metadata.py
Comment (OmarAI2003) [inline src/scribe_data/check/check_language_metadata.py:161]: Does this line  
`languages_in_metadata = {key.lower(): value for key, value in _languages.items()}`  
give all the languages in `_languages`, including the sublanguages?

Comment (catreedle) [reply src/scribe_data/check/check_language_metadata.py:161]: yes. but it doesn't lowercase all the sublanguages if any is mistakenly written, I just noticed.

Comment (OmarAI2003) [reply src/scribe_data/check/check_language_metadata.py:161]: There is also a helper function called `list_all_languages` at the bottom of `utils.py` that lists all the languages in the JSON file. You can make use of it here to check for sublanguages; it would be helpful.

Comment (catreedle) [reply src/scribe_data/check/check_language_metadata.py:161]: Thanks for the suggestion! Could you explain further how best to use it in this context? I want to include parent languages so I convert the languages directory to match the format of `language_metadata.json` for easier comparison. Since `list_all_languages` doesn’t show the parent languages, I’m unsure how to use it effectively for my comparison.

**Review by andrewtavis — APPROVED** [review]
I'm bringing this down and integrating the checks for isos and qids into the other workflows 😊 Thanks, @catreedle! :) Really great to have the support on the checks 🚀

**Comment by catreedle** [comment]
Thank you! @andrewtavis

**Pull request merged**
